cfme/tests/ansible/test_embedded_ansible_basic.py

```diff
@@ -416,3 +432,58 @@ def test_embedded_ansible_repository_playbook_link(ansible_repository):
     # Asserting 'PlaybookRepositoryView' which is navigated from Repository to Playbooks view.
     # Playbooks view is coming from Repository view which is not existing Playbooks view.
     assert view.is_displayed
+
+
+@pytest.mark.tier(2)
+@pytest.mark.meta(automates=[1656308])
+def test_embed_tower_repo_add_new_zone(appliance, ansible_repository, new_zone, request):
+    """
+    Test whether repository or credentials on New Zone.
+
+    Bugzilla:
+        1656308
+
+    Polarion:
+        assignee: sbulage
+        casecomponent: Ansible
+        initialEstimate: 1/2h
+        testSteps:
+            1. Configure a CFME appliance with the Embedded Ansible provider
+            2. Create a new zone
+            3. Move the appliance into the new zone
+            4. Add an embedded Ansible repository or credential
+        expectedResults:
+            1. Check Embedded Ansible Role is started.
+            2.
+            3.
+            4. Check Repository or Credentials were added.
+    """
+    # Check Ansible repository via fixture is created or not.
+    assert ansible_repository.exists
+    # Check Appliance is moved to newly created Zone.
+    assert new_zone.description == appliance.server.zone.description
+
+    # Creating new Repository after moving appliance to new zone.
+    repositories = appliance.collections.ansible_repositories
+    try:
+        playbooks_yaml = cfme_data.ansible_links.playbook_repositories
+        playbook_name = getattr(request, 'param', 'embedded_ansible')
+        repository = repositories.create(
+            name=fauxfactory.gen_alpha(),
+            url=getattr(playbooks_yaml, playbook_name),
+            description=fauxfactory.gen_alpha(),
+            scm_branch="second_playbook_branch"
+        )
+    except (KeyError, AttributeError):
+        message = "Missing ansible_links content in cfme_data, cannot setup repository"
+        logger.exception(message)  # log the exception for debug of the missing content
+        pytest.fail(message)
```

Comment:
Since the data is missing, I would expect a skip rather than a fail in this scenario

Reply:
You're right it should ideally skip test, but if I use `pytest.fail` then we should get the exact error or information why test failed which skip won't tell.

Reply:
@sbulage `skip` can pass information about why it skipped, just pass the `message` string in the same way your doing with `fail` here.

Reply:
@john-dupuy Agree that `skip` will also give information but in the logs. Also, this test is related to CRUD so I think failing rather skipping is more appropriate .?

Reply:
@sbulage While the test is crud, for it to fail, in my opinion, something must go wrong on the CFME side. In this case, nothing has gone wrong on the CFME side, it's just that our yamls don't have the `ansible_links` content. Because I this, I think a skip is more appropriate.

Reply:
Yup !!! as discussed offline, updating it. :smile: